### ISSUE TITLE
Cannot parse datatable with examples

### ISSUE THREAD

**Issue by sayurionishi** [pytest-dev/pytest-bdd#741]
Sample Scenario

```
Scenario Outline: User selects bookmark position
    When the user selects A in the bookmarks
    Then the position should be the following
      | x   | y   | z   |
      | <x> | <y> | <z> |
```

    Examples:
      | x | y | z |
      | 1 | 2 | 3 |

Before, it could parse the values correctly, returning 1, 2, 3, but now it returns the placeholders as strings `<x>, <y>, <z>.`
This worked before the recent change in data tables.
Hope you could help check this issue, thank you

**Comment by jsa34** [comment]
Hello @sayurionishi 

Could you please provide a full MVP?  

This looks invalid gherkin syntax - there is no Feature, for example.

Could you also provide the step definitions used?

**Comment by sayurionishi** [comment]
Good day, here it is @jsa34 
Feature: Bookmarks Tests
Scenario Outline: User selects bookmark position
    When the user selects A in the bookmarks
    Then the position should be the following
```
      | x   | y   | z   |
      | <x> | <y> | <z> |
```

    Examples:
      | x | y | z |
      | 1 | 2 | 3 |
      
I have also edited the first comment as it does not display the datatable properly      
pytest version: 8.1.1
pytest-bdd version: 8.0.0
OS:  Windows

Thank you

**Comment by youtux** [comment]
Interesting use case, I wonder if other implementations treat data tables as part of the template, or not. Maybe to be raised with the Cucumber guys.

Sounds like a valid use case for me though, hopefully it won't be too hard to fix.

**Comment by jsa34** [comment]
Agreed - I have asked to get a conclusive idea of where example parame should affect (for example, in Cucumber JVM, I know scenario names can use example params)

**Comment by vianney-g** [comment]
I faced the same problem.

Waiting for the fix, you can manually replace the datatable's placeholders in the step definitions with the examples' values supplied by the `_pytest_bdd_example` fixture.

**Comment by thburghout** [comment]
For future readers:

```
datatable = [[_pytest_bdd_example.get(x[1:-1], x) for x in row] for row in datatable]
```

**Comment by jsa34** [comment]
> For future readers:
> 
> ```python
> datatable = [[_pytest_bdd_example.get(x[1:-1], x) for x in row] for row in datatable]
> ```

This shouldn't be needed?

With version 8 onwards, you just need to use the step arg "datatable".  This is documented in the README

**Comment by youtux** [comment]
I just released pytest-bdd 8.1.0 with the fix 👍

**Comment by sayurionishi** [comment]
Thank you all for the help! 😊 @jsa34 @youtux
